jni/include/native_engines_stream_support.h

```diff
@@ -119,6 +121,110 @@ class NativeEngineIndexInputMediator {
 
 
 
+/**
+ * This class delegates the provided index output to do IO processing.
+ * In most cases, it is expected that IndexOutputWithBuffer was passed down to this,
+ * which eventually have Lucene's IndexOutput to write bytes.
+ */
+class NativeEngineIndexOutputMediator {
```

Comment:
If any member variables need to manage their own memory, can we consider using smart pointers instead of raw pointers.

Reply:
No member variables are allocated in memory at the moment by design. 
we don't need to use smart pointers!
Personally, I prefer to avoid using smart pointers as possible

Reply:
If this becuase of Heap vs Stack and due to performance hit ..I am ok !!